layout-server/src/test/java/org/finos/vuu/layoutserver/controller/ApplicationLayoutControllerTest.java

```diff
@@ -0,0 +1,75 @@
+package org.finos.vuu.layoutserver.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.finos.vuu.layoutserver.dto.response.ApplicationLayoutDto;
+import org.finos.vuu.layoutserver.model.ApplicationLayout;
+import org.finos.vuu.layoutserver.service.ApplicationLayoutService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.modelmapper.ModelMapper;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class ApplicationLayoutControllerTest {
+    private static ApplicationLayoutService mockService;
+    private static ApplicationLayoutController controller;
+    private static final ModelMapper modelMapper = new ModelMapper();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeEach
+    public void setup() {
+        mockService = Mockito.mock(ApplicationLayoutService.class);
+        controller = new ApplicationLayoutController(mockService, modelMapper);
+    }
+
+    @Test
+    public void getApplicationLayout_validUser_returnsLayoutFromService() throws JsonProcessingException {
+        String user = "user";
+        JsonNode definition = objectMapper.readTree("{\"id\":\"main-tabs\"}");
+
+        when(mockService.getApplicationLayout(user))
+                .thenReturn(new ApplicationLayout(user, definition));
+
+        ApplicationLayoutDto response = controller.getApplicationLayout(user);
+
+        assertThat(response.getUsername()).isEqualTo(user);
+        assertThat(response.getDefinition()).isEqualTo(definition);
+
+        verify(mockService, times(1)).getApplicationLayout(user);
+    }
+
+    @Test
+    public void createApplicationLayout_validUser_callsService() throws JsonProcessingException {
+        String user = "user";
+        JsonNode definition = objectMapper.readTree("{\"id\":\"main-tabs\"}");
+
+        controller.createApplicationLayout(user, definition);
+
+        verify(mockService, times(1)).createApplicationLayout(user, definition);
+    }
+
+    @Test
+    public void updateApplicationLayout_validUser_callsService() throws JsonProcessingException {
+        String user = "user";
+        JsonNode definition = objectMapper.readTree("{\"id\":\"main-tabs\"}");
+
+        controller.updateApplicationLayout(user, definition);
+
+        verify(mockService, times(1)).updateApplicationLayout(user, definition);
+    }
+
+    @Test
+    public void deleteApplicationLayout_validUser_callsService() {
```

Comment:
aren't we calling the service deleteApplication regardless of the user being valid or not?

Reply:
We are. I've put `validUser` on all the test names just to satisfy the convention (`methodName_stateUnderTest_expectedBehaviour`). All the controller does is pass through to the service, so the tests aren't the most useful.

Reply:
discussed offline. See whether we can find a different naming convention that fits better otherwise remove _stateUnderTest_

Reply:
I've gone with `anyUsername`/`anyInput`: 4c2b809e.